### DISCUSSION TITLE
cross-compiling for resource-constrained targets

### DISCUSSION THREAD

**Discussion by biosbob** [ASDAlexander77/TypeScriptCompiler#146] [Ideas]
interesting project....

my use-case targets resource-constrained embedded MCUs with ~32K of memory, which today are programmed in "bare-metal" C code....

were i to "constrain" my use of typescript to require a very minimal runtime, could i in theory use `tsc` with any target supported by LLVM???

**Comment by ASDAlexander77** [comment]
in short answer: in theory - yes, In practice I didn't try many different "targets". The only targets which I tried were "x86, x64, Wasm32, Wasm64" and I am sure "arm64" will work.  any 32-bit platforms required extra testing.
